packages/effect/src/internal/rateLimiter.ts

```diff
@@ -0,0 +1,74 @@
+// Direct port of https://github.com/svroonland/rezilience/blob/master/rezilience/shared/src/main/scala/nl/vroste/rezilience/RateLimiter.scala
+import type { Duration } from "effect"
+import { Chunk, Deferred, Effect, pipe, Queue, Ref, Stream } from "effect"
+
+// Find the next power of 2 greater than or equal to n.
+/** @internal */
+const nextPow2 = (n: number): number => {
```

Comment:
This method is duplicated and also exists in `pubsub.ts`. What would be best location for utility functions like this one?

Reply:
a file in `internal` probably is best

Reply:
Done.